test/variable.jl

```diff
@@ -219,6 +219,19 @@ end
         @test_throws ErrorException z[end]
     end
 
+    @testset "Key iterators of JuMPContainers" begin
+        m = Model()
+        @variable(m, x[2:3, 1:2])
+        @variable(m, y[2:3])
+        @variable(m, z[-1:1,[:red,"blue"]])
+        @variable(m, w[i=1:2,j=1:2 ; (j+i) % 2 == 1])
+
+        @test vec(collect(keys(x))) == [(2,1), (3,1), (2,2), (3,2)]
```

Comment:
Nit: The whitespace doesn't follow the JuMP style guide (there should be spaces after commas). However, not a big deal for the release-0.18 branch.